context/app/static/js/components/detailPage/visualization/VisualizationWrapper/VisualizationError.jsx

```diff
@@ -1,49 +1,48 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import { ErrorBoundary as SentryErrorBoundary } from '@sentry/react';
+import { faro } from '@grafana/faro-web-sdk';
 
 import { DetailPageSection } from 'js/components/detailPage/style';
 import { SpacedSectionButtonRow } from 'js/shared-styles/sections/SectionButtonRow';
+import DetailsAccordion from 'js/shared-styles/accordions/DetailsAccordion';
 import { StyledSectionHeader } from '../Visualization/style';
 import { VisualizationErrorBoundaryBackground } from './style';
 
-// consider using "react-error-boundary" package in the future to convert this to a functional component
-class VisualizationErrorBoundary extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { hasError: false, error: null };
-  }
-
-  componentDidCatch(error, errorInfo) {
-    // You can also log the error to an error reporting service
-    console.error(error, errorInfo);
-    this.setState({
-      hasError: true,
-      error: { error, errorInfo },
-    });
-  }
-
-  render() {
-    const { hasError, error } = this.state;
-    const { isPublicationPage, uuid, shouldDisplayHeader, children } = this.props;
-
-    if (hasError) {
+function VisualizationFallback({ isPublicationPage, uuid, shouldDisplayHeader }) {
+  return useCallback(
+    ({ message, componentStack }) => {
       return (
         <DetailPageSection id={isPublicationPage ? `visualization-${uuid}` : 'visualization'}>
           <SpacedSectionButtonRow
             leftText={shouldDisplayHeader ? <StyledSectionHeader>Visualization</StyledSectionHeader> : undefined}
           />
           <VisualizationErrorBoundaryBackground>
             <div>The Vitessce visualization encountered an error. Please try again or contact support.</div>
-            <details>
-              <summary>Click to expand error details</summary>
-              <div>{error.errorInfo.componentStack}</div>
-            </details>
+            <DetailsAccordion summary="Click to expand error details">
+              {message} {componentStack}
+            </DetailsAccordion>
           </VisualizationErrorBoundaryBackground>
         </DetailPageSection>
       );
-    }
+    },
+    [isPublicationPage, uuid, shouldDisplayHeader],
+  );
+}
 
-    return children;
-  }
+function VisualizationErrorBoundary({ children, ...rest }) {
+  return (
+    <SentryErrorBoundary
+      beforeCapture={(scope) => {
+        scope.setTag('location', 'vitessce');
+      }}
+      onError={(error) => {
+        faro.logError('VisualizationErrorBoundary', { error });
+      }}
+      fallback={VisualizationFallback({ ...rest })}
```

Comment:
I wonder if we could combine the props on lines 11 and 13 then change this line to
```
fallback={error => <VisualizationFallback {...error} {...rest} />}
```

This seems like it should resolve the concern we discussed the other week re: how `useCallback` doesn't actually resolve the underlying unstable component issue that lint throws.

We can revisit this in a separate ticket instead, though.